core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -234,10 +264,24 @@ private void processScanResult(Snapshot.Key key, Scan scan, TransactionResult re
           snapshot.getId());
     }
 
-    // We always update the read set to create before image by using the latest record (result)
-    // because another conflicting transaction might have updated the record after this
-    // transaction read it first.
-    putIntoReadSetInSnapshot(key, Optional.of(result));
+    Optional<TransactionResult> ret = Optional.of(result);
+    if (!scan.getConjunctions().isEmpty()) {
+      // Check if the result matches the conjunctions
```

Comment:
Ditto.

```suggestion
      // Because we also get records whose before images match the conjunctions, we need to check if
      // the current status of the records actually match the conjunctions.
```

Reply:
Fixed in 4ec25d631eab49f358a7068b7af68ac854602648. Thanks!